site/content/en/docs/manual/basics/attach-cloud-storage.md

```diff
@@ -8,376 +8,415 @@ description: 'Instructions on how to attach cloud storage using UI'
-object data select protect object data type. When all the information is entered click `Create` to create the bucket.
+1. Prepare data.
+   For more information,
+   see [prepare the dataset](#prepare-the-dataset).
+2. Go to container and click **Upload**.
+3. Click **Browse for files** and select images.
+   > Note: If images are in folder, specify folder in the **Advanced settings** > **Upload to folder**.
+4. Click **Upload**.
 
-![](/images/google_cloud_storage_tutorial1.jpg)
+![](/images/azure_blob_container_tutorial9.jpg)
 
-### Upload
+### SAS token
 
-Prepare the dataset as in the point [prepare dataset](#prepare-dataset).
+Use the SAS token to grant secure access to the container.
 
-To upload files, you can simply drag and drop files and folders into a browser window
-or use the `upload folder` and/or `upload files`.
+To configure the SAS token:
 
-### Access permissions
+1. Go to **Home** > **Resourse groups** > You resourse name > Your storage account.
+2. On the left menu, click **Shared access signature**.
+3. Change the following fields:
+   - **Allowed services**: Enable **Blob** . Disable all other fields.
+   - **Allowed resource types**: Enable **Container** and **Object**. Disable all other fields.
+   - **Allowed permissions**: Enable **Read** and **List**. Disable all other fields.
+   - **Start and expiry date**: Set up start and expiry dates.
+   - **Allowed protocols**: Select **HTTPS and HTTP**
+   - Leave all other fields with default parameters.
+4. Click **Generate SAS token** and copy **SAS token**.
 
-To access Google Cloud Storage from CVAT you will need a `Project ID`
-you can find it by going to [cloud resource manager page](https://console.cloud.google.com/cloud-resource-manager)
+![](/images/azure_blob_container_tutorial3.jpg)
 
-![](/images/google_cloud_storage_tutorial5.jpg)
+### Personal use
 
-#### Create a service account and key file
+For personal use, you can use the **Access Key**
+from your storage account in the CVAT **SAS Token** field.
 
-To access your bucket you need a key file and a service account. To create a service account,
-go to `IAM & Admin`/`Service Accounts` and press `Create Service Account`. Enter your account
-name and click `Create And Continue`. Select a role for example `Basic`/`Viewer`.
-Next, you can give access rights to the service account, to complete click `Done`.
+To get the **Access Key**:
 
-![](/images/google_cloud_storage_tutorial2.jpg)
+1. In the Azure, go to the **Security + networking** > **Access Keys**
+2. Click **Show** and copy the key.
 
-The account you created will appear in the service accounts list, open it and go to the `Keys` tab.
-To create a key, click `ADD` and select `Create new key`, next you need to choose the key type `JSON` and select `Create`.
-The key file will be downloaded automatically.
+![](/images/azure_blob_container_tutorial8.jpg)
 
-![](/images/google_cloud_storage_tutorial3.jpg)
+### Attach Azure storage
```

Comment:
```suggestion
### Attach Azure Blob Container
```
I guess we need to call it so because there is the `File shares` service that can also be considered as a `storage`